plugins/messagelayer/plugin.go

```diff
@@ -155,6 +155,7 @@ func Tangle() *tangle.Tangle {
 			}),
 			tangle.SyncTimeWindow(Parameters.TangleTimeWindow),
 			tangle.StartSynced(Parameters.StartSynced),
+			tangle.ForceCacheTime(database.Parameters.ForceCacheTime),
```

Comment:
Instead of passing an int, initialize the `database.CacheTimeProvider` instance in a plugin either here or create a singleton in `plugins/database` and import it.

Reply:
but this breaks the pattern...
I don't understand what's wrong with the current code

Reply:
Yes, theoretically, singletons are bad, but the whole plugins framework is built on top of them, and creating one more singleton for the `CacheTimeProvider` doesn't break the pattern, but it keeps the code consistent. When we are ready we will refactor the plugin framework and make it use the DI. 

When you pass an int to the `tangle.New()` it makes the function do one more thing: initialize `database.CacheTimeProvider` instance and it violates the single responsibility pattern.
It's better to initialize all Go objects at the beginning of the program and then pass them instead of raw values.

Reply:
ok I agree